### DISCUSSION TITLE
react-native: While trying to resolve module `mobx-react` from file `\App.tsx`, the package `\node_modules\mobx-react\package.json` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`\node_modules\mobx-react\dist\mobxreact.esm.js`. Indeed, none of these files exist:

### DISCUSSION THREAD

**Discussion by alexeyAK1** [mobxjs/mobx#2602] [Q&A] [answered]
Hi, I'm developing on react, I decided to try react-native. I took a bunch of react-native and mobx. But that something went wrong. And at first everything worked correctly, then I decided to add react-native-push-notification to the project and everything fell. Tried to roll back, but to no avail. If you have any ideas, I would be very grateful.

"scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint . --ext .js,.jsx,.ts,.tsx"
  },
  "dependencies": {
    "mobx": "^6.0.3",
    "mobx-react": "^7.0.2",
    "react": "16.13.1",
    "react-native": "0.63.3"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/runtime": "^7.8.4",
    "@react-native-community/eslint-config": "^1.1.0",
    "@types/jest": "^25.2.3",
    "@types/react-native": "^0.63.2",
    "@types/react-test-renderer": "^16.9.2",
    "@typescript-eslint/eslint-plugin": "^2.27.0",
    "@typescript-eslint/parser": "^2.27.0",
    "babel-jest": "^25.1.0",
    "eslint": "^6.5.1",
    "jest": "^25.1.0",
    "metro-react-native-babel-preset": "^0.59.0",
    "react-test-renderer": "16.13.1",
    "typescript": "^3.8.3"
  },

```
import {observer} from 'mobx-react';
import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

declare const global: {HermesInternal: null | {}};

const App = observer(() => {
  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={styles.scrollView}>
          <Header />
          {global.HermesInternal == null ? null : (
            <View style={styles.engine}>
              <Text style={styles.footer}>Engine: Hermes</Text>
            </View>
          )}
          <View style={styles.body}>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>Step One</Text>
              <Text style={styles.sectionDescription}>
                Edit <Text style={styles.highlight}>App.tsx</Text> to change
                this screen and then come back to see your edits.
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>See Your Changes</Text>
              <Text style={styles.sectionDescription}>
                <ReloadInstructions />
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>Debug</Text>
              <Text style={styles.sectionDescription}>
                <DebugInstructions />
              </Text>
            </View>
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionTitle}>Learn More</Text>
              <Text style={styles.sectionDescription}>
                Read the docs to discover what to do next:
              </Text>
            </View>
            <LearnMoreLinks />
          </View>
        </ScrollView>
      </SafeAreaView>
    </>
  );
});

const styles = StyleSheet.create({
  scrollView: {
    backgroundColor: Colors.lighter,
  },
  engine: {
    position: 'absolute',
    right: 0,
  },
  body: {
    backgroundColor: Colors.white,
  },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  highlight: {
    fontWeight: '700',
  },
  footer: {
    color: Colors.dark,
    fontSize: 12,
    fontWeight: '600',
    padding: 4,
    paddingRight: 12,
    textAlign: 'right',
  },
});

export default App;
```

**Comment by danielkcz** [comment, marked as answer]
Thank you for reporting this. It was a small mistake in yesterday's migration chores to monorepo. Please update mobx-react to 7.0.4 and it should work.

**Comment by alexeyAK1** [comment]
Thank you for your prompt response, but now it returns the following error (react-native doesn't seem to use react-dom): **Error: Unable to resolve module `react-dom` from `node_modules\mobx-react-lite\dist\mobxreactlite.esm.js`: react-dom could not be found within the project.**

**Reply by danielkcz** [reply]
You are right, forgot about this part when migrating. Should be fixed in mobx-react-lite 3.1.5. You might need to force an update of `mobx-react` which has it as a transitive dependency.

**Reply by alexeyAK1** [reply]
Thank you very much for the prompt solution of problems!!!
